src/cljs/andel/core.cljs

```diff
@@ -56,15 +59,18 @@
     {:width (.-width (.measureText ctx s)) :height line-h}))
 
 (defn make-editor-state []
-  (let [ch (a/chan)]
-    {:text (text/make-text "")
-     :selection [49 4956]
-     :caret {:offset 0 :v-col 0}
-     :lexer-broker ch
-     :modespec "text/x-java"
-     :timestamp 0
-     :lines []
-     :first-invalid 0}))
+  {:document {:text (text/make-text "")
+              :lexer-broker (a/chan)
+              :modespec "text/x-java"
+              :timestamp 0
+              :lines []
+              :first-invalid 0}
+   :editor {:caret {:offset 0 :v-col 0}
+            :selection [0 0]}
+   :viewport {:pos [0 0]
```

Comment:
I think that metrics also should be part of editor state(viewport part).
Because it depends on font that is part of editor configuration.